Build failure: qpdfview

### Steps To Reproduce

Steps to reproduce the behavior:
1. Checkout nixpkgs master (currently 1ac0d12079c)
2. `nix build .#qpdfview`

### Build log

```
qmakePrePhase
qtPreHook
unpacking sources
unpacking source archive /nix/store/vyn11644n95g947igvr4f5qjjah10hz9-qpdfview-0.5.tar.gz
source root is qpdfview-0.5
setting SOURCE_DATE_EPOCH to timestamp 1674902447 of file qpdfview-0.5/translations/qpdfview_zh_TW.ts
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
QMAKEPATH=/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev:/nix/store/76slzawvhdhps3d4mn73i4zl3cd6h8id-qtwayland-5.15.9-dev:/nix/store/2vx1a4n2fswyzcgwz9x09nv1wa35jqxz-qtdeclarative-5.15.9-dev:/nix/s>
qmake PREFIX=/nix/store/rrs3bq87dg299f7q3ybbpz4s2jxiamz2-qpdfview-0.5.0 NIX_OUTPUT_OUT=/nix/store/rrs3bq87dg299f7q3ybbpz4s2jxiamz2-qpdfview-0.5.0 NIX_OUTPUT_DEV=/nix/store/rrs3bq87dg299f7q3ybbpz4s2jxiamz2-qpdfvie>
Info: creating stash file /build/qpdfview-0.5/.qmake.stash
qmake: enabled parallel building
qmake: enabled parallel installing
building
build flags: -j30 SHELL=/nix/store/51sszqz1d9kpx480scb1vllc00kxlx79-bash-5.2-p15/bin/bash
( test -e Makefile.pdf-plugin || /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/bin/qmake -o Makefile.pdf-plugin /build/qpdfview-0.5/pdf-plugin.pro PREFIX=/nix/store/rrs3bq87dg299f7q3ybbpz4s2jxiamz>
make[1]: Entering directory '/build/qpdfview-0.5'
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -pipe -O2 -std=gnu++11 -Wall -Wextra -dM -E -o moc-pdf/moc_predefs.h /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/mkspecs/features/data/dummy.cpp
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/bin/moc -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_>
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/bin/moc -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_>
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/bin/moc -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_>
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/bin/moc -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DHAS_POPPLER_14 -DHAS_POPPLER_18 -DHAS_POPPLER_20 -DHAS_POPPLER_22 -DHAS_POPPLER_24 -DHAS_POPPLER_26 -DHAS_POPPLER_31 -DHAS_POPPLER_35 -DHAS_POPPLER>
In file included from /nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-qt5.h:53,
                 from sources/annotationwidgets.cpp:39:
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:639:5: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https://>
  639 |     QLinkedList<QPointF> linePoints() const;
      |     ^~~~~~~~~~~
In file included from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qvarlengtharray.h:43,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qmetatype.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qobject.h:54,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qwidget.h:45,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qframe.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qabstractscrollarea.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qtextedit.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qplaintextedit.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/QPlainTextEdit:1,
                 from sources/annotationwidgets.h:25,
                 from sources/annotationwidgets.cpp:22:
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:640:30: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  640 |     void setLinePoints(const QLinkedList<QPointF> &points);
      |                              ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:854:11: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  854 |     QList<QLinkedList<QPointF>> inkPaths() const;
      |           ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:855:34: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  855 |     void setInkPaths(const QList<QLinkedList<QPointF>> &paths);
      |                                  ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
In file included from /nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-qt5.h:53,
                 from sources/pdfmodel.cpp:39:
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:639:5: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https://>
  639 |     QLinkedList<QPointF> linePoints() const;
      |     ^~~~~~~~~~~
In file included from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qvarlengtharray.h:43,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qmetatype.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qobject.h:54,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcoreapplication.h:46,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/QCoreApplication:1,
                 from sources/pdfmodel.h:26,
                 from sources/pdfmodel.cpp:24:
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:640:30: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  640 |     void setLinePoints(const QLinkedList<QPointF> &points);
      |                              ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:854:11: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  854 |     QList<QLinkedList<QPointF>> inkPaths() const;
      |           ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:855:34: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  855 |     void setInkPaths(const QList<QLinkedList<QPointF>> &paths);
      |                                  ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
In file included from sources/pdfmodel.cpp:47:
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:6: error: 'optional' in namespace 'std' does not name a template type
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      |      ^~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:1: note: 'std::optional' is only available from C++17 onwards
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      | ^~~
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qmetatype.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qobject.h:54,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcoreapplication.h:46,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/QCoreApplication:1,
                 from sources/pdfmodel.h:26,
                 from sources/pdfmodel.cpp:24:
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:640:30: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  640 |     void setLinePoints(const QLinkedList<QPointF> &points);
      |                              ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:854:11: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  854 |     QList<QLinkedList<QPointF>> inkPaths() const;
      |           ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:855:34: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  855 |     void setInkPaths(const QList<QLinkedList<QPointF>> &paths);
      |                                  ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
In file included from sources/pdfmodel.cpp:47:
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:6: error: 'optional' in namespace 'std' does not name a template type
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      |      ^~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:1: note: 'std::optional' is only available from C++17 onwards
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      | ^~~
In file included from /nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:47,
                 from sources/formfieldwidgets.cpp:27:
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:639:5: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https://>
  639 |     QLinkedList<QPointF> linePoints() const;
      |     ^~~~~~~~~~~
In file included from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qvarlengtharray.h:43,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qmetatype.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qobject.h:54,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qsharedpointer_impl.h:65,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qsharedpointer.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtGui/qpixmap.h:48,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtGui/qicon.h:46,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qabstractbutton.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/qcheckbox.h:44,
                 from /nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtWidgets/QCheckBox:1,
                 from sources/formfieldwidgets.h:25,
                 from sources/formfieldwidgets.cpp:22:
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:640:30: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  640 |     void setLinePoints(const QLinkedList<QPointF> &points);
      |                              ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:854:11: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  854 |     QList<QLinkedList<QPointF>> inkPaths() const;
      |           ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-annotation.h:855:34: warning: 'template<class T> class QLinkedList' is deprecated: Use std::list instead [8;;https:/>
  855 |     void setInkPaths(const QList<QLinkedList<QPointF>> &paths);
      |                                  ^~~~~~~~~~~
/nix/store/bfxhrxlbabg79z85mlyai9favwvjiqgf-qtbase-5.15.9-dev/include/QtCore/qcontainerfwd.h:51:26: note: declared here
   51 | template <class T> class QLinkedList;
      |                          ^~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:6: error: 'optional' in namespace 'std' does not name a template type
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      |      ^~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-form.h:888:1: note: 'std::optional' is only available from C++17 onwards
  888 | std::optional<CryptoSignBackend> POPPLER_QT5_EXPORT activeCryptoSignBackend();
      | ^~~
sources/pdfmodel.cpp: In member function 'virtual qpdfview::Model::Properties qpdfview::Model::PdfDocument::properties() const':
sources/pdfmodel.cpp:1132:30: warning: 'void Poppler::Document::getPdfVersion(int*, int*) const' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecate>
 1132 |     m_document->getPdfVersion(&pdfMajorVersion, &pdfMinorVersion);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/lhxp2q66iq9m21npb68mx2gw7z7k20hc-poppler-qt5-23.07.0-dev/include/poppler/qt5/poppler-qt5.h:1658:28: note: declared here
 1658 |     Q_DECL_DEPRECATED void getPdfVersion(int *major, int *minor) const;
      |                            ^~~~~~~~~~~~~
make[1]: *** [Makefile.pdf-plugin:496: objects-pdf/formfieldwidgets.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile.pdf-plugin:490: objects-pdf/pdfmodel.o] Error 1
make[1]: Leaving directory '/build/qpdfview-0.5'
make: *** [Makefile:48: sub-pdf-plugin-pro-make_first-ordered] Error 2
```

### Additional context

This seems to be directly caused by 170b6714ad98897c95ba26b58e1e693e1274a903 introduced in #241731

No update of `qpdfview` is availble upstream.

### Notify maintainers

@jtojnar (author of the `poppler` PR)
@7c6f434c (`qpdfview` maintainer)

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.36, NixOS, 23.11 (Tapir), 23.11.20230710.277e7de`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - channels(root): `"nixos, nixpkgs"`
 - channels(wogan): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/office/qpdfview/default.nix
+++ b/pkgs/applications/office/qpdfview/default.nix
@@ -52,6 +52,11 @@ mkDerivation rec {
     "APPDATA_INSTALL_PATH=${placeholder "out"}/share/appdata"
   ];
 
+  env = {
+    # Fix build due to missing `std::option`.
+    NIX_CFLAGS_COMPILE = "-std=c++17";
+  };
+
   meta = with lib; {
     description = "A tabbed document viewer";
     license = licenses.gpl2Plus;

